packages/arbor-store/src/Arbor.test.ts

```diff
@@ -136,4 +138,62 @@ describe("Arbor", () => {
       expect(store.root).toEqual([{ name: "User 2" }, { name: "User 3" }])
     })
   })
+
+  describe("custom data model", () => {
+    class Todo extends Model<Todo> {
+      id: string
+      text: string
+      completed = false
+
+      complete() {
+        this.completed = true
+      }
+
+      activate() {
+        this.completed = false
+      }
+
+      get status() {
+        return this.completed ? "Completed" : "Active"
+      }
+    }
+
+    it("supports user defined data models", () => {
+      const store = new Arbor([
+        new Todo({ text: "Do the dishes", completed: false }),
+        new Todo({ text: "Clean the house", completed: true }),
```

Comment:
Plugins such as `LocalStorage` would have to be aware of these user-defined data models so they may deserialize data accordingly.